services/director-v2/tests/unit/dynamic_scheduler/test_modules_dynamic_sidecar_scheduler__core2__workflow_runner.py

```diff
@@ -0,0 +1,149 @@
+# pylint: disable=protected-access
+# pylint: disable=redefined-outer-name
+
+import logging
+from typing import Any
+from unittest.mock import AsyncMock
+
+import pytest
+from pytest import LogCaptureFixture
+from simcore_service_director_v2.modules.dynamic_sidecar.scheduler._core2._action import (
+    Action,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.scheduler._core2._context_base import (
+    ContextInterface,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.scheduler._core2._marker import (
+    mark_step,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.scheduler._core2._models import (
+    ActionName,
+    StepName,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.scheduler._core2._workflow import (
+    Workflow,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.scheduler._core2._workflow_context import (
+    WorkflowContext,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.scheduler._core2._workflow_runner import (
+    _iter_index_step,
+    workflow_runner,
+)
+
+logger = logging.getLogger(__name__)
+
+
+# FIXTURES
+
+
+@pytest.fixture
+async def workflow_context(
+    context: ContextInterface,
+) -> WorkflowContext:
+    workflow_context = WorkflowContext(
+        context=context, app=AsyncMock(), workflow_name="unique", action_name="first"
+    )
+    await workflow_context.setup()
+    yield workflow_context
+    await workflow_context.teardown()
+
+
+# TESTS
+
+
+async def test_iter_index_step():
```

Comment:
this is more of a dev test to check somthing, ... I would decorate as ``pytest.mark.skip(reason="DEV")`` and save time in CI

Reply:
So this means this basically will never be ran again, right?
It can be disabled, even though the saved time is basically not noticeable `0.01` seconds.

Reply:
another way is to create a marker in the ``setup.cfg`` (e.g. ``dev``)  and mark it as ``@pytest.mark.dev``. then in the CI script we make sure to include ``no-dev`` to filter them out.

It is not about this specific test but at the accumulation of them. I think these tests are handy while developing but I tend to delete them (well, i copy them into my sandbox repo as a reminder) or skip.